server/src/test/java/org/solarbank/server/unit/ControllerExceptionHandlerTest.java

```diff
@@ -0,0 +1,124 @@
+package org.solarbank.server.unit;
+
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.solarbank.server.ControllerExceptionHandler;
+import org.solarbank.server.ErrorMessage;
+import org.solarbank.server.ValidationMessage;
+import org.solarbank.server.dto.CalculateRequest;
+import org.solarbank.server.dto.ErrorResponse;
+import org.springframework.core.MethodParameter;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ControllerExceptionHandlerTest {
+
+    @InjectMocks
+    private ControllerExceptionHandler handler;
+
+    @Mock
+    private BindingResult bindingResult;
+
+    @Mock
+    private MethodArgumentNotValidException ex;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    public void bindingResultException_validationExceptionHandler_expectedErrorResponse() throws NoSuchMethodException {
```

Comment:
The below code will throw a NoSuchMethodException. I can't figure out why this would be as surely the setter is created by lombok? unless its due to test environment being different. Not sure if this is a problem as including `throws NoSuchMethodException` allows the test to run. 'Method method = CalculateRequest.class.getMethod("setPanelEfficiency", Double.class);'